modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -740,6 +750,38 @@ private Object _getValue(
 		}
 	}
 
+	private Object _handleComplexPropertyExpressionPropertyExpression(
+		PropertyExpression propertyExpression, List<String> acc) {
+
+		if (propertyExpression instanceof ComplexPropertyExpression) {
+			ComplexPropertyExpression complexPropertyExpression =
+				(ComplexPropertyExpression)propertyExpression;
+
+			acc.add(propertyExpression.getName());
+
+			PropertyExpression complexPropertyExpressionPropertyExpression =
+				complexPropertyExpression.getPropertyExpression();
+
+			return _handleComplexPropertyExpressionPropertyExpression(
+				complexPropertyExpressionPropertyExpression, acc);
+		}
+
+		if (propertyExpression instanceof CollectionPropertyExpression) {
```

Comment:
I think it would be easier to read/understand if we include:

```
if (
[...]
} else if {
[...]
} else {
[...]
```

Reply:
Done